docs/src/defining_examples.md

```diff
@@ -0,0 +1,279 @@
+# Defining a bijector: examples
+
+Here we provide two different worked examples of defining a custom bijector.
+
+## Cyclic permutation
+
+We start with something simple: a bijector that performs a cyclic permutation of the elements of a vector.
+
+```@example cyclic
+using Bijectors
+
+struct CircShift <: Bijector
+    shift::Int
+end
+```
+
+As described in the [previous page](./defining.md), the only function you absolutely _must_ implement is `with_logabsdet_jacobian`.
+
+Let's think for a moment about what the Jacobian is.
+`CircShift` is a mapping from `ℝⁿ → ℝⁿ` that permutes the elements of the input vector.
+For example, `CircShift(1)` would map a length-3 vector `x = [x1, x2, x3]` to `y = [x3, x1, x2]`.
+
+That means that the Jacobian matrix is
+
+$$J = \begin{bmatrix}
+\partial y_1/\partial x_1 & \partial y_1/\partial x_2 & \partial y_1/\partial x_3 \\
+\partial y_2/\partial x_1 & \partial y_2/\partial x _2 & \partial y_2/\partial x_3 \\
+\partial y_3/\partial x_1 & \partial y_3/\partial x_2 & \partial y_3/\partial x_3
+\end{bmatrix}
+= \begin{bmatrix}
+0 & 0 & 1 \\
+1 & 0 & 0 \\
+0 & 1 & 0
+\end{bmatrix}.$$
+
+In general, the Jacobian of such a transformation is a [permutation matrix](https://en.wikipedia.org/wiki/Permutation_matrix).
+The determinant of a permutation matrix is either `1` or `-1`, depending on whether the permutation is even or odd.
+This means that the log-absolute determinant of the Jacobian is always `0`.
+
+!!! note
+    
+    In our case, the permutation is always even, since we are performing a cyclic shift. However, since we only need the log-*absolute* determinant, we don't even need to worry about knowing this.
+
+We can now implement `with_logabsdet_jacobian`.
+
+```@example cyclic
+function Bijectors.with_logabsdet_jacobian(
```

Comment:
Bijectors doesn't actually own `with_logabsdet_jacobian`, and additionally it might be nice to show that you don't have to depend on Bijectors if you only want to define such a simple bijector:


```suggestion
function ChangesOfVariables.with_logabsdet_jacobian(
```